### PR TITLE
caffe2 | MSVS compatibility fixes

### PR DESCRIPTION
**Pull request by boguscoder**
Summary: Code changes required to build caffe2 for windows with toolchain used by FB.

Differential Revision: D13953258

### CODE REVIEW

**Comment by boguscoder** [comment]
can anyone with write perm. re-trigger two failed checks, please, they look intermittent

File: caffe2/operators/quantized/int8_utils.h [hunk 1]
```diff
@@ -143,7 +143,11 @@ activationLimits(float scale, int32_t zero_point, Activation Ac) {
       return {QuantizeUint8(scale, zero_point, 0.0),
               std::numeric_limits<uint8_t>::max()};
     default:
+#ifdef _MSC_VER
+      break;
```
Comment (peterjc123) [inline caffe2/operators/quantized/int8_utils.h]: The equivalent should be __assume(0).

**Comment by boguscoder** [comment]
@peterjc123 where shall I seek help with tests failed, there are only few of them but they dont seem to be related to my changes

**Pull request closed without merging**